java/src/main/java/com/codurance/training/tasks/refactored/service/ViewService.java

```diff
@@ -0,0 +1,19 @@
+package com.codurance.training.tasks.refactored.service;
+
+import com.codurance.training.tasks.refactored.Task;
+
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+
+public interface ViewService {
+    void show(Map<String, List<Task>> tasks, PrintWriter out);
+
+    void tasksCreatedDate(String createdDate ,Map<String, List<Task>> tasks, PrintWriter out);
+
+    void tasksDeadLineDate(String createdDate, Map<String, List<Task>> tasks, PrintWriter out);
+
+    void tasksByProject(String DeadLineDate, Map<String, List<Task>> tasks, PrintWriter out);
+
+    void taskDueToday(Map<String, List<Task>> tasks, PrintWriter out);
+}
```

Comment:
This violates the interface seggregation principle... like if any class wants to implements this interface then class will be forced to override all of these methods..

Also if we want to add a new feature for example viewAllTasks() then we need to add that method in this interface that violates the open close principle..

One approach can be seggregating this methods to seperate interfaces and implementing seperate classes each feature

other approach could be having one interface with one common method for all view features implemented by respective classes... so in this if we want to add new feature then we just need to create a new class which just implements the existing interface.